graph/src/ext/futures.rs

```diff
@@ -1,13 +1,13 @@
 use futures::sync::oneshot;
-use std::sync::{Arc, Mutex};
-use tokio::prelude::{Future, Poll, Stream};
+use std::sync::{Arc, Mutex, Weak};
+use tokio::prelude::{future::Fuse, Future, Poll, Stream};
 
 /// A cancelable stream or future.
 ///
 /// It can be canceled through the corresponding `CancelGuard`.
```

Comment:
A couple other details would be useful here, how about "This type wraps a stream or future to make it cancelable." and a hint about creating these using `.cancelable` in the FutureExtension/StreamExtension?